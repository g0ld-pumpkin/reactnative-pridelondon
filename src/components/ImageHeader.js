// @flow
import React from "react";
import { ImageBackground, StyleSheet } from "react-native";
import type { ImageRef } from "../data/image-ref";
import ContentPadding from "./ContentPadding";
import Text from "./Text";
import { lightNavyBlueColor, whiteColor } from "../constants/colors";

type Props = {
  image: ImageRef,
  title: string | string[],
  subtitle?: string
};

const ImageHeader = ({ image, title, subtitle }: Props) => (
  <ImageBackground style={styles.image} source={image} resizeMode="cover">
    <ContentPadding
      padding={{
        small: { horizontal: 8, vertical: 8 },
        medium: { horizontal: 16, vertical: 16 },
        large: { horizontal: 16, vertical: 16 }
      }}
      style={styles.contentPadding}
    >
      {(Array.isArray(title) ? title : [title]).map((line, index, all) => {
        const zIndex = all.length - index;
        return (
          <Text
            key={line}
            color="lightNavyBlueColor"
            type="h1"
            style={[{ zIndex }, styles.title]}
          >
            {line}
          </Text>
        );
      })}
      {subtitle ? (
        <Text type="h2" color="whiteColor" style={styles.subtitle}>
          {subtitle}
        </Text>
      ) : null}
    </ContentPadding>
  </ImageBackground>
);

ImageHeader.defaultProps = {
  subtitle: ""
};

const styles = StyleSheet.create({
  image: {
    aspectRatio: 2 / 1,
    justifyContent: "flex-end"
  },
  contentPadding: {
    alignItems: "flex-start"
  },
  title: {
    backgroundColor: whiteColor,
    marginTop: -8,
    paddingHorizontal: 8,
    paddingTop: 8
  },
  subtitle: {
    backgroundColor: lightNavyBlueColor,
    paddingHorizontal: 8,
    paddingTop: 4
  }
});

export default ImageHeader;
