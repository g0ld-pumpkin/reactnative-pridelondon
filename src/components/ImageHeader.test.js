// @flow
import React from "react";
import { shallow } from "enzyme";
import ImageHeader from "./ImageHeader";

it("renders correctly", () => {
  const output = shallow(<ImageHeader image={0} title="Fruit" />);
  expect(output).toMatchSnapshot();
});

it("renders correctly with multiple lines", () => {
  const output = shallow(
    <ImageHeader image={0} title={["This is a", "longer heading"]} />
  );
  expect(output).toMatchSnapshot();
});

it("renders correctly with subtitle", () => {
  const output = shallow(
    <ImageHeader image={0} title="Fruit" subtitle="Is very good" />
  );
  expect(output).toMatchSnapshot();
});
