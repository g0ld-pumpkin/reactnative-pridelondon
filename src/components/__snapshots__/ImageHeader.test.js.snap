// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ImageBackground
  resizeMode="cover"
  source={0}
  style={
    Object {
      "aspectRatio": 2,
      "justifyContent": "flex-end",
    }
  }
>
  <ContentPadding
    padding={
      Object {
        "large": Object {
          "horizontal": 16,
          "vertical": 16,
        },
        "medium": Object {
          "horizontal": 16,
          "vertical": 16,
        },
        "small": Object {
          "horizontal": 8,
          "vertical": 8,
        },
      }
    }
    style={
      Object {
        "alignItems": "flex-start",
      }
    }
  >
    <Text
      color="lightNavyBlueColor"
      key="Fruit"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
            "marginTop": -8,
            "paddingHorizontal": 8,
            "paddingTop": 8,
          },
        ]
      }
      type="h1"
    >
      Fruit
    </Text>
  </ContentPadding>
</ImageBackground>
`;

exports[`renders correctly with multiple lines 1`] = `
<ImageBackground
  resizeMode="cover"
  source={0}
  style={
    Object {
      "aspectRatio": 2,
      "justifyContent": "flex-end",
    }
  }
>
  <ContentPadding
    padding={
      Object {
        "large": Object {
          "horizontal": 16,
          "vertical": 16,
        },
        "medium": Object {
          "horizontal": 16,
          "vertical": 16,
        },
        "small": Object {
          "horizontal": 8,
          "vertical": 8,
        },
      }
    }
    style={
      Object {
        "alignItems": "flex-start",
      }
    }
  >
    <Text
      color="lightNavyBlueColor"
      key="This is a"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Array [
          Object {
            "zIndex": 2,
          },
          Object {
            "backgroundColor": "#ffffff",
            "marginTop": -8,
            "paddingHorizontal": 8,
            "paddingTop": 8,
          },
        ]
      }
      type="h1"
    >
      This is a
    </Text>
    <Text
      color="lightNavyBlueColor"
      key="longer heading"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
            "marginTop": -8,
            "paddingHorizontal": 8,
            "paddingTop": 8,
          },
        ]
      }
      type="h1"
    >
      longer heading
    </Text>
  </ContentPadding>
</ImageBackground>
`;

exports[`renders correctly with subtitle 1`] = `
<ImageBackground
  resizeMode="cover"
  source={0}
  style={
    Object {
      "aspectRatio": 2,
      "justifyContent": "flex-end",
    }
  }
>
  <ContentPadding
    padding={
      Object {
        "large": Object {
          "horizontal": 16,
          "vertical": 16,
        },
        "medium": Object {
          "horizontal": 16,
          "vertical": 16,
        },
        "small": Object {
          "horizontal": 8,
          "vertical": 8,
        },
      }
    }
    style={
      Object {
        "alignItems": "flex-start",
      }
    }
  >
    <Text
      color="lightNavyBlueColor"
      key="Fruit"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
            "marginTop": -8,
            "paddingHorizontal": 8,
            "paddingTop": 8,
          },
        ]
      }
      type="h1"
    >
      Fruit
    </Text>
    <Text
      color="whiteColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "backgroundColor": "#2d2f7f",
          "paddingHorizontal": 8,
          "paddingTop": 4,
        }
      }
      type="h2"
    >
      Is very good
    </Text>
  </ContentPadding>
</ImageBackground>
`;
